code-challenges.test.js

```diff
@@ -18,43 +18,145 @@
+// Might be able to mab through the objects, access the values, and capitalize + slice(1) all in one go. That would be ideal
+// Once the names are capitalized the array will be mapped through again and the return will be a string that uses string interpolation
+// it will look something like `${object.value(name)} is a ${object.value(occupation)}.` I am assuming. I will need to google to get syntax correct
+
+const canName = (array) => {
+ let newArray = array.map(value => {
+   return value.name.split(' ')
+})
+ let capName = []
+ for(let i=0; i<newArray.length; i++){
+   for( let j=0; j<newArray[i].length; j++){
+     return newArray[i][j].charAt(0).toUpperCase() + newArray[i][j].slice(1)
+   }
+   // capName.push(capValue)
+
+   // return capName
+ }}
+
+
+// This problem has taken days and hours and I still can't seem to get it. I was able to pull out the names and create an array with an array of names.
+// I looped through each array to capitalize the names and it just capitalizes the first array and I can't seem to figure out why it is stopping. As I am out of time, I have to let it go and see what my classmates have come up with. 
+
 
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-var hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+describe("remainder", () => {
+  test("the function should return an array that contains the remainders of the numbers in the original array when they are divided by 3.", () => {
+    var hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    var hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+    expect(remainder(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(remainder(hodgepodge2)).toEqual([ 2, 1, -1 ])
+
+  })
+})
+
+// var hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
-var hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+// var hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 
 
 // b) Create the function that makes the test pass.
 
+// Create a function called remainder
+// Takes a parameter of an array
+// Create an empty array called numOnly
+// Iterate through the array and if the type is equal to a number then that number will be pushed into numOnly
+// Iterate through numOnly (probably using map) and for each index I will divide by 3 and print the remainder in the returned array.
+
+const remainder = (array) => {
+  let numOnly = []
+  for(let i=0; i<array.length; i++){
+    if( typeof array[i] === "number"){
+      numOnly.push(array[i])
+    }
+  }
+   let remainArr = numOnly.map(value => {
+    return value % 3
+  })
+  return remainArr
+}
```

Comment:
good stuff! Question though? why use a for loop and push into an array instead of using .filter()?

Reply:
I honestly didn't think about it. I am so used to using for loops that I just default to it. I should probably start to google other iterations to practice.